beacon-chain/sync/utils.go

```diff
@@ -30,9 +30,21 @@ func (s sortedObj) Len() int {
 // sort the provided blocks and roots in ascending order. This method assumes that the size of
 // block slice and root slice is equal.
 func (r *Service) sortBlocksAndRoots(blks []*ethpb.SignedBeaconBlock, roots [][32]byte) ([]*ethpb.SignedBeaconBlock, [][32]byte) {
+	// Remove duplicate blocks received
```

Comment:
I believe this should be in its own function, say, `dedupBlocksAndRoots()`, then in beacon-blocks-by-range we just have:
```golang
	blks, roots = r.dedupBlocksAndRoots(blks, roots)
	blks, roots = r.sortBlocksAndRoots(blks, roots)
```
Otherwise, it is not clear why sorting dedups as well.